torchtune/models/convert_weights.py

```diff
@@ -231,6 +230,7 @@ def _permute(t, n_heads):
 
 def tune_to_peft_adapter_config(
     adapter_config: Dict[str, Any],
+    base_model_name_or_path: Optional[str] = None,
```

Comment:
@ebsmothers  How was the PEFT model figuring this out before?

Reply:
Oh it wasn't. That's why we loaded PEFT models in two steps: 

```
model = AutoModelForCausalLM.from_pretrained(model_id)
peft_model = PeftModel.from_pretrained(model, checkpoint_dir)
```

instead of 

```
AutoModelForCausalLM.from_pretrained(checkpoint_dir)
```

I had a hacky version of this in #2026 but it was pointed out by @pbontrager that this shouldn't be present for models that are gonna get pushed to the hub (in that case we would want the hub model ID here, not a local path). I'm curious how we wanna address that here, should we wrap hugging-face CLI upload with something that will modify the path?

Reply:
> peft_model = PeftModel.from_pretrained(model, checkpoint_dir)

I don't think that this works either. The info has to be inside of the adapter_config. Maybe if input_dir == output_dir?